packages/atomic/src/components/atomic-icon/atomic-icon.tsx

```diff
@@ -0,0 +1,44 @@
+import {Component, Element, h, Host, Prop} from '@stencil/core';
+import {parseAssetURL} from '../../utils/utils';
+
+/**
+ * The `atomic-icon` component displays an SVG icon with a 1:1 aspect ratio.
+ *
+ * This component can either display an icon from the list of available icons, a direct link or an SVG element.
+ */
+@Component({
+  tag: 'atomic-icon',
+  styleUrl: 'atomic-icon.pcss',
+  shadow: false,
+  assetsDirs: ['assets'],
+})
+export class AtomicIcon {
+  @Element() host!: HTMLElement;
+
+  /**
+   * Specifies the icon to display.
+   *
+   * When the icon begins with http://, https://, ./ or ../, it will be fetched and displayed.
+   * When the icon begins with assets://, it will be displayed from the list of available icons.
+   * Otherwise, the icon will be displayed directly as an SVG element.
+   */
+  @Prop() icon!: string;
+
+  private svg: string | null = null;
+
+  private async fetchIcon(url: string) {
+    const response = await fetch(url);
+    return response.status === 200 || response.status === 304
+      ? await response.text()
+      : null;
```

Comment:
should we leave it as `null` or throw some sort of error or log something like `icon could not be fetched from ${url}`

Reply:
Hmm, it could be helpful. Does the `atomic-icon` also use the `atomic-component-error` component to show a message in the UI?

Reply:
It doesn't, I'll try to make it display errors.